lib/galaxy/webapps/base/api.py

```diff
@@ -100,9 +96,14 @@ async def __call__(self, scope: "Scope", receive: "Receive", send: "Send") -> No
                     raise RuntimeError(f"File at path {self.path} is not a file.")
 
         # This is where we diverge from the superclass, this adds support for byte range requests
+        if not scope["method"].upper() == "HEAD" and self.send_header_only:
+            # Not a head request, but nginx_x_accel_redirect_base / send_header_only, we don't send a body
+            self.send_header_only = True
+            self.headers["content-length"] = "0"
+
         start = 0
         end = stat_result.st_size - 1
-        if not self.send_header_only:
+        if not scope["method"].upper() == "HEAD":
```

Comment:
```suggestion
        if not send_header_only:
```

and need to drop line 62 and change line 124 from `if self.send_header_only:` to `if send_header_only:`